eras/alonzo/impl/src/Cardano/Ledger/Alonzo/Plutus/Context.hs

```diff
@@ -193,3 +210,78 @@ type family PlutusTxInfo (l :: Language) where
   PlutusTxInfo 'PlutusV1 = PV1.TxInfo
   PlutusTxInfo 'PlutusV2 = PV2.TxInfo
   PlutusTxInfo 'PlutusV3 = PV3.TxInfo
+
+-- | This is just like `mkPlutusScript`, except it is guaranteed to be total through the enforcement
+-- of support by the type system and `EraPlutusTxInfo` type class instances for supported plutus
+-- versions.
+mkSupportedPlutusScript ::
+  forall l era.
+  (HasCallStack, EraPlutusTxInfo l era) =>
+  Plutus l ->
+  PlutusScript era
+mkSupportedPlutusScript plutus =
+  case mkPlutusScript plutus of
+    Nothing ->
+      error $
+        "Impossible: "
+          ++ show plutus
+          ++ " language version should be supported by the "
+          ++ eraName @era
+    Just plutusScript -> plutusScript
+
+-- | This is just like `mkPlutusScript`, except it is guaranteed to be total through the enforcement
+-- of support by the type system and `EraPlutusTxInfo` type class instances for supported plutus
```

Comment:
```suggestion
-- | This is just like `mkBinaryPlutusScript`, except it is guaranteed to be total through the enforcement
-- of support by the type system and `EraPlutusTxInfo` type class instances (via calling `mkSupportedPlutusScript) for supported plutus
```

I guess you meant `mkBinaryPlutusScript` if I understand the change correctly, right? If so, rest of the comment could be slightly modified as well to avoid confusion (I added a slight suggestion for that as well but feel free to avoid and rewrite completely if you think that would be better).